core/domain/suggestion_jobs_one_off_test.py

```diff
@@ -0,0 +1,979 @@
+                            'needs_update': True
+                        }
+                    }
+                }
+            },
+            'interaction': {
+                'answer_groups': [answer_group],
+                'confirmed_unclassified_answers': [],
+                'customization_args': {
+                    'choices': {
+                        'value': [html_content]
+                    },
+                    'showChoicesInShuffledOrder': {
+                        'value': True
+                    }
+                },
+                'default_outcome': {
+                    'dest': None,
+                    'feedback': {
+                        'content_id': 'feedback_2',
+                        'html': html_content
+                    },
+                    'param_changes': [],
+                    'refresher_exploration_id': None,
+                    'labelled_as_correct': True,
+                    'missing_prerequisite_skill_id': None
+                },
+                'hints': [{
+                    'hint_content': {
+                        'content_id': 'hint_1',
+                        'html': 'Hint 1'
+                    }
+                }],
+                'solution': {
+                    'answer_is_exclusive': False,
+                    'correct_answer': 0,
+                    'explanation': {
+                        'content_id': 'solution',
+                        'html': '<p>This is a solution.</p>'
+                    }
+                },
+                'id': 'MultipleChoiceInput'
+            },
+            'param_changes': [],
+            'solicit_answer_details': False,
+            'classifier_model_id': None
+        }
+        suggestion_dict_with_math = {
+            'suggestion_id': 'skill2.thread1',
+            'suggestion_type': suggestion_models.SUGGESTION_TYPE_ADD_QUESTION,
+            'target_type': suggestion_models.TARGET_TYPE_SKILL,
+            'target_id': 'skill2',
+            'target_version_at_submission': 1,
+            'status': suggestion_models.STATUS_ACCEPTED,
+            'author_name': 'author',
+            'final_reviewer_id': self.reviewer_id,
+            'change': {
+                'cmd': question_domain.CMD_CREATE_NEW_FULLY_SPECIFIED_QUESTION,
+                'question_dict': {
+                    'question_state_data': question_state_dict_with_math,
+                    'language_code': 'en',
+                    'question_state_data_schema_version': (
+                        feconf.CURRENT_STATE_SCHEMA_VERSION),
+                    'linked_skill_ids': ['skill_2']
+                },
+                'skill_id': 'skill_2',
+                'skill_difficulty': 0.3,
+            },
+            'score_category': 'question.skill1',
+            'last_updated': utils.get_time_in_millisecs(self.fake_date)
+        }
+        with self.swap(
+            feedback_models.GeneralFeedbackThreadModel,
```

Comment:
WHy we need this?

Reply:
We need a valid thread id to be created in order for a suggestion to be submitted.
Pls refer to the tests in `suggestions_services_test` , It's similar to those tests.